src/huggingface_hub/README.md

```diff
@@ -236,6 +241,43 @@ Examples using the `commit` context manager:
 ...     torch.save(model.state_dict(), "model.pt")
   ```
 
+### Non-blocking behavior
+
+The pushing methods have access to a `blocking` boolean parameter to indicate whether the push should happen
+asynchronously.
+
+In order to see if the push has finished or its status code (to spot a failure), one should use the `command_queue`
+property on the `Repository` object.
+
+For example:
+
+```python
+from huggingface_hub import Repository
+
+repo = Repository("<local_folder>", clone_from="<user>/<model_name>")
+
+with repo.commit("Commit message", blocking=False):
+    # Save data
+
+last_command = repo.command_queue[-1]
+
+# Status of the push command
+last_command.status  
+# Will return the status code
+#     -> "running" will indicate the push is still ongoin
+#     -> 0 will indicate the push has completed successfully
+#     -> non-zero code indicates the error code if there was an error
+
+# if there was an error, the stderr may be inspected
+last_command.stderr
```

Comment:
no strong feeling, but since we have a `is_done` convenience method, maybe we could have a `failed` method that just checks if there's an error.

Reply:
Done!